pypsa-to-iamc

```diff
@@ -0,0 +1,634 @@
+            
+            #Emissions : Oil
+ 
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Liquids|Fossil'] = 0.001*ct*(n.links_t.p2.filter(like ='oil boiler').filter(like =country)).sum().sum() 
+            
+
+            
+            ###Energy generation: total,peaks,and percentiles
+            
+            sheet_var={"Yearly_generation_supply":'Generation|Yearly',"Winter_peak_generation":'Generation|Winter peak',"summer_peak_generation":'Generation|Summer peak',\
+                       "Percentile50_generation":'Generation|Percentile 50',"Percentile25_generation":'Generation|Percentile 25',\
+                       'Percentile_25_consumpt':'Hourly power consumption|Percentile 25','Percentile_50_consumpt':'Hourly power consumption|Percentile 50','Summer_peak_consumpt' :'Hourly power consumption|Summer peak',\
+                       'Winter_peak_consumpt':'Hourly power consumption|Winter peak'}
+            
+            ct=3        # Hourly resolution 
+        
+            for sheet in sheets.keys(): 
+             if sheet in ("Yearly_generation_supply","Winter_peak_generation","summer_peak_generation","Percentile50_generation","Percentile25_generation"):
+              sh=sheets[sheet]
+               
+              #Energy generation : Solar PV (rooftop, ground)   
+            
+              var[sh][sheet_var[sheet]+'|Electricity|Solar|Rooftop PV'] = 0.001*ct*gen_des(time(n.generators_t.p.filter(like ='solar rooftop').filter(like =country).sum(axis=1)))
+              var[sh][sheet_var[sheet]+'|Heat|Solar'] = 0.001*ct*gen_des(time(n.generators_t.p.filter(like ='solar thermal').filter(like =country).sum(axis=1)))
+              var[sh][sheet_var[sheet]+'|Electricity|Solar|Open field'] = 0.001*ct*gen_des(time(n.generators_t.p.filter(like ='solar').filter(like =country).sum(axis=1)))\
+              -var[sh][sheet_var[sheet]+'|Heat|Solar']-var[sh][sheet_var[sheet]+'|Electricity|Solar|Rooftop PV']
+              var[sh][sheet_var[sheet]+'|Electricity|Solar'] =var[sh][sheet_var[sheet]+'|Electricity|Solar|Open field']+var[sh][sheet_var[sheet]+'|Electricity|Solar|Rooftop PV']
+             
+              #Energy generation :  onshore and offshore wind   
+             
+              var[sh][sheet_var[sheet]+'|Electricity|Wind|Onshore']=0.001*ct*gen_des(time(n.generators_t.p.filter(like ='onwind').filter(like =country).sum(axis=1))) 
+              var[sh][sheet_var[sheet]+'|Electricity|Wind|Offshore']=0.001*ct*gen_des(time(n.generators_t.p.filter(like ='offwind').filter(like =country).sum(axis=1))) 
+              var[sh][sheet_var[sheet]+'|Electricity|Wind']=var[sh][sheet_var[sheet]+'|Electricity|Wind|Onshore']+var[sh][sheet_var[sheet]+'|Electricity|Wind|Offshore']
+             
+              #Energy generation : Nuclear
+            
+              var[sh][sheet_var[sheet]+'|Electricity|Nuclear'] =0.001*ct*gen_des(time((n.links_t.p1.filter(like ='nuclear').filter(like =country)).sum(axis=1)))
```

Comment:
make sure that all the generated energy is positive as discussed in the meeting

Reply:
Done